quickstart/aspnet-core/ContosoTeamStats/RedisConnection.cs

```diff
@@ -142,21 +142,26 @@ private async Task ForceReconnectAsync(bool initializing = false)
                 _firstErrorTime = DateTimeOffset.MinValue;
                 _previousErrorTime = DateTimeOffset.MinValue;
 
-                if (_connection != null)
+                // Create a new connection
+                ConnectionMultiplexer _newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
+
+                // Swap current connection with the new connection
+                ConnectionMultiplexer _oldConnection = Interlocked.Exchange(ref _connection, _newConnection);
+
+                if (_oldConnection != null)
                 {
                     try
                     {
-                        await _connection.CloseAsync();
+                        await _oldConnection.CloseAsync();
                     }
                     catch
                     {
                         // Ignore any errors from the old connection
                     }
                 }
 
-                Interlocked.Exchange(ref _connection, null);
-                ConnectionMultiplexer newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
-                Interlocked.Exchange(ref _connection, newConnection);
+                // Dispose old connection
```

Comment:
Tiny: "Dispose" caught my eye as unexpected since there isn't a `using` block or direct call to `Dispose()`. Would it be OK to say something like "Clear old connection"?

Reply:
Ohh, shoot, I forgot. The `_oldConnection.CloseAsync()` calls `Dispose()` behind the scenes. So I think it's right that this is indeed allowing the `_oldConnection` to be garbage collected.

Reply:
Ohh yes. Good point. Along those lines, I believe `_oldConnection` can be renamed to `oldConnection` since it's a local variable?

Reply:
Done